addons/website/tests/test_website_reset_password.py

```diff
@@ -75,3 +75,23 @@ def test_02_multi_user_login(self):
         # The most specific user should be selected
         self.authenticate("bobo@mail.com", "bobo@mail.com")
         self.assertEqual(self.session["uid"], user2.id)
+
+    def test_multi_website_reset_password_user_specific_user_account(self):
+        # Create same user on different websites with 'Specific User Account'
+        # option enabled and then reset password. Only the user from the
+        # current website should be reset.
+        website_1, website_2 = self.env['website'].create([
+            {'name': 'Website 1', 'specific_user_account': True},
+            {'name': 'Website 2', 'specific_user_account': True},
+        ])
+
+        login = 'user@example.com' # same login for both users
+        user_1, user_2 = self.env["res.users"].with_context(no_reset_password=True).create([
+            {"website_id": website_1.id, "login": login, "email": login, 'name': login},
+            {"website_id": website_2.id, "login": login, "email": login, 'name': login},
+        ])
+
+        self.env['res.users'].with_context(website_id=website_1.id).reset_password(login)
+
+        self.assertTrue(user_1.signup_valid)
+        self.assertFalse(user_2.signup_valid)
```

Comment:
Maybe you want to do that check before the call to reset_password to check both are `assertFalse` to make the test stronger (so you are sure you are coming from the state you expect)